src/octotree.js

```diff
@@ -24,9 +25,23 @@ $(document).ready(() => {
       if (event.target === window) layoutChanged();
     });
 
-    $toggler.click(toggleSidebarAndSave);
+    $pinner.click(togglePinAndSave);
+
+    $sidebar.on('keyup mouseleave mousemove', debounce({
```

Comment:
I think this piece of logic is not right, causing the bugs I reported.

I think the logic could be like this:

* mouseleave: if (float mode) cause hiding timer to start (say, hide after 3s)
* mouseenter: remove any hiding timer
* keyup: reset any hiding timer (i.e. if keeps typing and there's a timer, restart it) 
* When toggle pin to true, remember to delete hiding timer.
* No need debouncing (unless mouseenter doesn't work in a certain edge case and we have to use mousemove; keyup shouldn't be a problem as it won't be that fast and a timer reset isn't too much work, prefer simpler code even if it's not as efficient, as long as it doesn't cause visible performance degrade).